### DISCUSSION TITLE
How can I make my classes provide explanations for failed assertions?

### DISCUSSION THREAD

**Discussion by MaddyGuthridge** [pytest-dev/pytest#10492] [Q&A] [answered]
I understand that it is possible to give custom assertion messages by defining a `pytest_assertrepr_compare` in the `conftest.py` for every project that uses these classes, but I would much rather define it once in the package I am distributing and have it be automatically used in the project's dependents without requiring any action from them. Is there a way I can cleanly do this?

**Comment by The-Compiler** [comment, marked as answer]
You could [ship a pytest plugin](https://docs.pytest.org/en/7.2.x/how-to/writing_plugins.html#setuptools-entry-points) with your project, which implements that hook.

**Reply by MaddyGuthridge** [reply]
Thanks, I got it working!
